src/librustc/dep_graph/graph.rs

```diff
@@ -388,10 +388,7 @@ impl DepGraph {
             |_| None,
             |data, key, fingerprint, _| {
                 let mut current = data.borrow_mut();
-                let krate_idx = current.node_to_node_index[
-                    &DepNode::new_no_params(DepKind::Krate)
```

Comment:
`eval_always` queries are hardcoded to depend on the entire HIR (`Krate`) here. I changed these to instead always execute instead of executing only when the HIR changes. Also see comments / perf on https://github.com/rust-lang/rust/pull/59091